enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/RequireJavaVersion.java

```diff
@@ -76,6 +76,7 @@ public void execute(EnforcerRuleHelper helper) throws EnforcerRuleException {
                 + " Build: " + detectedJdkVersion.getBuildNumber() + " Qualifier: "
                 + detectedJdkVersion.getQualifier());
 
+        super.setMessage(prepareMessage(detectedJdkVersion, getVersion()));
```

Comment:
We override message which can be provided by rule configuration - is it intentionally?

Reply:
Hmm. So would it make sense to override the message only if it has been not provided by the rule configuration? Otherwise I am not sure how else would be get the additional info in.

Reply:
simply set if current message is null